ad7768-1fmcz/ad77681.c

```diff
@@ -322,12 +635,306 @@ int32_t ad77681_set_power_mode(struct ad77681_dev *dev,
+					      AD77681_REG_DIGITAL_FILTER,
+					      AD77681_DIGI_FILTER_DEC_RATE_MSK,
+					      AD77681_DIGI_FILTER_DEC_RATE(decimate));
+		/* SINC3*/
+	} else {
+		uint8_t sinc3_LSB = 0, sinc3_MSB = 0;
+
+		sinc3_MSB = sinc3_osr >> 8;
+		sinc3_LSB = sinc3_osr & 0x00FF;
+
+		ret |= ad77681_spi_write_mask(dev,
+					      AD77681_REG_DIGITAL_FILTER,
+					      AD77681_DIGI_FILTER_FILTER_MSK,
+					      AD77681_DIGI_FILTER_FILTER(filter));
+
+		ret |= ad77681_spi_write_mask(dev,
+					      AD77681_REG_SINC3_DEC_RATE_MSB,
+					      AD77681_SINC3_DEC_RATE_MSB_MSK,
+					      AD77681_SINC3_DEC_RATE_MSB(sinc3_MSB));
+
+		ret |= ad77681_spi_write_mask(dev,
+					      AD77681_REG_SINC3_DEC_RATE_LSB,
+					      AD77681_SINC3_DEC_RATE_LSB_MSK,
+					      AD77681_SINC3_DEC_RATE_LSB(sinc3_LSB));
+	}
+
+	if ( ret == SUCCESS) {
+		dev->decimate = decimate;
+		dev->filter = filter;
+		/* Sync pulse after each filter change */
+		ret |= ad77681_initiate_sync(dev);
+	}
+
+	return ret;
+}
+
+/**
+ * Enable 50/60 Hz rejection
+ * @param dev   -  The device structure.
+ * @param enable - The positive reference buffer selector
+ * 					Accepted values: true
+ *									 false
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_set_50HZ_rejection(struct ad77681_dev *dev,
+				   uint8_t enable)
+{
+	int32_t ret;
+
+	ret = ad77681_spi_write_mask(dev,
+				     AD77681_REG_DIGITAL_FILTER,
+				     AD77681_DIGI_FILTER_60HZ_REJ_EN_MSK,
+				     AD77681_DIGI_FILTER_60HZ_REJ_EN(enable));
+
+	return ret;
+}
+
+/**
+ * Set the REF- reference buffer
+ * @param dev - The device structure.
+ * @param REFn - The negative reference buffer selector
+ * 					Accepted values: AD77681_BUFn_DISABLED
+ *									 AD77681_BUFn_ENABLED
+ *									 AD77681_BUFn_FULL_BUFFER_ON
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_set_continuos_read(struct ad77681_dev *dev,
+				   enum ad77681_continuous_read continuous_enable)
+{
+	int32_t ret;
 
-	dev->mclk_div = clk_div;
+	if (continuous_enable) {
+		ret = ad77681_spi_write_mask(dev,
+					     AD77681_REG_INTERFACE_FORMAT,
+					     AD77681_INTERFACE_CONT_READ_MSK,
+					     AD77681_INTERFACE_CONT_READ_EN(continuous_enable));
+	} else {
+		/* To exit the continuous read mode, a key 0x6C must be
+		written into the device over the SPI*/
+		uint8_t end_key = EXIT_CONT_READ;
+		ret = spi_write_and_read(dev->spi_desc, &end_key, 1);
+	}
+
+	return ret;
+}
+
+/**
+ * Power down / power up the device
+ * @param dev -			The device structure.
+ * @param sleep_wake	Power down, or power up the ADC
+ * 						Accepted values: AD77681_SLEEP
+ *										 AD77681_WAKE
+ * @return 0 in case of success, negative error code otherwise.
+ */
+
+int32_t ad77681_power_down(struct ad77681_dev *dev,
+			   enum ad77681_sleep_wake sleep_wake)
+{
+	int32_t ret;
+
+	if (sleep_wake == AD77681_SLEEP) {
+		ret = ad77681_spi_reg_write(dev, AD77681_REG_POWER_CLOCK,
```

Comment:
the value of `ret` is not handled in this function.
Replace `return 0` with `return ret` at the end of the function

Reply:
Done fixing